packages/eslint-plugin-query/src/rules/no-rest-desctructuring/no-rest-destructuring.utils.ts

```diff
@@ -0,0 +1,11 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/utils'
+import type { TSESTree } from '@typescript-eslint/utils'
+
+export const NoRestDestructuringUtils = {
+  isObjectRestDestructuring(node: TSESTree.Node): boolean {
```

Comment:
I would rename it to hasRestElement and move it to ASTUtils, but that's totally fine as well.

Reply:
I think hasRestElement would be confused with the array rest destructuring `[a, ...rest]`, but I'm okay with moving it to ASTUtils.